### PR TITLE
Add the schema admin api

### PR DESCRIPTION
**Pull request by congbobo184**
### Motivation
Continue the issues of #4782

### Verifying this change
Add the tests for it

Does this pull request potentially affect one of the following parts:
If yes was chosen, please highlight the changes

Dependencies (does it add or upgrade a dependency): (no)
The public API: (no)
The schema: (yes)
The default values of configurations: (no)
The wire protocol: (no)
The rest endpoints: (no)
The admin cli options: (no)
Anything that affects deployment: (no)

### Documentation
Does this pull request introduce a new feature? (yes / no)
If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
If a feature is not applicable for documentation, explain why?
If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Comment by congbobo184** [comment]
run Integration Tests

**Comment by congbobo184** [comment]
run java8 tests

**Comment by congbobo184** [comment]
run java8 tests

**Comment by congbobo184** [comment]
run java8 tests

**Comment by congbobo184** [comment]
run  Integration Tests

**Comment by congbobo184** [comment]
run Integration Tests

**Comment by congbobo184** [comment]
run Integration Tests

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/SchemasResource.java [hunk 1]
```diff
@@ -183,6 +218,34 @@ private static void handleGetSchemaResponse(AsyncResponse response,
...
+            if (isNull(schemas)) {
+                response.resume(Response.status(Response.Status.NOT_FOUND).build());
+            } else if (schemas.size() == 0) {
+                response.resume(Response.status(Response.Status.NOT_FOUND).build());
+            } else {
+                response.resume(
+                        Response.ok()
+                                .encoding(MediaType.APPLICATION_JSON)
+                                .entity(GetAllVersionsSchemaResponse.builder().getSchemaResponses(
+                                        schemas.stream().map(schemaAndMetadata -> GetSchemaResponse.builder()
```
Comment (sijie) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/SchemasResource.java]: Can you extract the `schemaAndMetadata` to `GetSchemaResponse` transformation into a common method? I think it should be reused across this class.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/SchemasResource.java [hunk 2]
```diff
@@ -291,6 +354,121 @@ public void postSchema(
...
+            )
+                    PostSchemaPayload payload,
+            @QueryParam("authoritative") @DefaultValue("false") boolean authoritative,
+            @Suspended final AsyncResponse response
+    ) {
+        validateDestinationAndAdminOperation(tenant, namespace, topic, authoritative);
+
+        String schemaId = buildSchemaId(tenant, namespace, topic);
+
+        SchemaCompatibilityStrategy schemaCompatibilityStrategy =SchemaCompatibilityStrategy
```
Comment (sijie) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/SchemasResource.java]:
```suggestion
        SchemaCompatibilityStrategy schemaCompatibilityStrategy = SchemaCompatibilityStrategy
```

**Comment by congbobo184** [comment]
> overall looks good. can you add tests for these endpoints?

OK, I will add tests and fix these problems from your comment

**Comment by congbobo184** [comment]
run cpp tests

**Comment by congbobo184** [comment]
run Integration Tests

**Comment by congbobo184** [comment]
run Integration Tests

**Comment by congbobo184** [comment]
run Integration Tests

**Comment by wolfstudy** [comment]
Change the Milestone to 2.4.2, because of conflict.

**Pull request merged**